### PR TITLE
Add ac001 keyboard

### PR DESCRIPTION
**Pull request by jonylee1986**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hfdkb/ac001/config.h [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#define SPI_SCK_PIN B3
+#define SPI_MOSI_PIN B5
+#define SPI_MISO_PIN B4
+#define SPI_MOSI_PAL_MODE 5
+
+#define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN C12
+#define WEAR_LEVELING_BACKING_SIZE (8 * 1024)
+
+#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#define RGB_MATRIX_CENTER {112,32}
```
Comment (zvecr) [inline keyboards/hfdkb/ac001/config.h:50]:
```suggestion
```

File: keyboards/hfdkb/ac001/rules.mk [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = no
```
Comment (zvecr) [inline keyboards/hfdkb/ac001/rules.mk:15]:
```suggestion
```

File: keyboards/hfdkb/ac001/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT( /* Base */
+		RGB_MOD, KC_LCTL, KC_ENT,KC_NO,KC_NO),
+    [1] = LAYOUT(
+		RGB_MOD, KC_LCTL, KC_ENT,KC_NO,KC_NO),
+    [2] = LAYOUT(
+		RGB_MOD, KC_LCTL, KC_ENT,KC_NO,KC_NO),
+    [3] = LAYOUT(
+		RGB_MOD, KC_LCTL, KC_ENT,KC_NO,KC_NO)
```
Comment (zvecr) [inline keyboards/hfdkb/ac001/keymaps/default/keymap.c:27]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Also,
* `readme.md` missing - needs to follow [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)

**Review by drashna — APPROVED** [review]

File: keyboards/hfdkb/ac001/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# ac001
+
+This is an enter key with rgb .
+
+* Keyboard Maintainer: [jonylee@hfd](https://github.com/jonylee1986)
+* Hardware Supported: ac001
+
+Make example for this keyboard (after setting up your build environment):
+
+    make hfdkb/ac001:default
```
Comment (waffle87) [inline keyboards/hfdkb/ac001/readme.md:10]:
```suggestion
    make hfdkb/ac001:default

Flashing example for this keyboard:

    make hfdkb/ac001:default:flash

```

**Review by waffle87 — APPROVED** [review]

**Pull request merged**